modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/RustServerCodegen.java

```diff
@@ -528,9 +528,8 @@ public CodegenOperation fromOperation(String path, String httpMethod, Operation
         // if "consumes" is defined (per operation or using global definition)
         if (consumes != null && !consumes.isEmpty()) {
             List<Map<String, String>> c = new ArrayList<Map<String, String>>();
-            for (String key : consumes) {
+            for (String mimeType : consumes) {
                 Map<String, String> mediaType = new HashMap<String, String>();
-                String mimeType = processMimeType(key);
 
                 if (mimeType.startsWith("Application/Xml")) {
```

Comment:
I suspect there may be problems - it looks as if later bits of code assume that the `mimetype` has been generated by `processMimeType` - hence this checking against `Application/Xml`.

According to the [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types), mime types are supposed to be case insensitive, but are generally lower case.

Now that we've no longer got `processMimeType`, do we need to be a bit more clever about how we detect the XML mime type?

Otherwise, it looks sensible.

Reply:
I think canonicalizing to lowercase will do.